src/arithmetic/curves.rs

```diff
@@ -62,6 +67,7 @@ pub trait CurveExt:
     ///     (g * x + &(h * r)).to_affine()
     /// }
     /// ```
+    #[cfg(feature = "alloc")]
     fn hash_to_curve<'a>(domain_prefix: &'a str) -> Box<dyn Fn(&[u8]) -> Self + 'a>;
```

Comment:
Technically it's not correct for a feature to enable a trait method for an exposed trait. Idea: make the whole `CurveExt` trait conditional on `std`. Only `halo2` right now needs `hash_to_curve` etc.

Reply:
I moved `CurveExt` behind `alloc`, and `CurveAffine` behind `std`.